localstack/packages/__init__.py

```diff
@@ -0,0 +1,349 @@
+import abc
+import functools
+import logging
+import os
+import threading
+from abc import ABC
+from enum import Enum
+from inspect import getmodule
+from typing import Callable, List, Optional
+
+from plugin import Plugin, PluginManager, PluginSpec
+
+from localstack import config
+from localstack.utils.platform import in_docker, is_debian, is_redhat
+from localstack.utils.run import run
+
+LOG = logging.getLogger(__name__)
+
+PLUGIN_NAMESPACE = "localstack.packages"
+
+
+class PackageException(Exception):
+    """Basic exception indicating that a package-specific exception occurred."""
+
+    pass
+
+
+class NoSuchVersionException(PackageException):
+    """Exception indicating that a requested installer version is not available / supported."""
+
+    pass
+
+
+class SystemNotSupportedException(PackageException):
+    """Exception indicating that the current system is not allowed."""
+
+    pass
+
+
+class InstallTarget(Enum):
+    """
+    Different installation targets.
+    Attention:
+    - These targets are directly used in the LPM API and are therefore part of a public API!
+    - The order of the entries in the enum define the default lookup order when looking for package installations.
+
+    These targets refer to the directories in config#Directories.
+    - VAR_LIBS: Used for packages installed at runtime. They are installed in a host-mounted volume.
+                This directory / these installations persist across multiple containers.
+    - STATIC_LIBS: Used for packages installed at build time. They are installed in a non-host-mounted volume.
+                   This directory is re-created whenever a container is recreated.
+    """
+
+    VAR_LIBS = config.dirs.var_libs
+    STATIC_LIBS = config.dirs.static_libs
```

Comment:
It definitely isn't necessary, a resolution path would work as well.
I just wanted to limit the possibilities here, since it was quite a hassle to introduce these two installation targets (which also have specific semantics).
This could be done fairly easy by creating an "install resolution path", defining the allowed values for LPM in the CLI interface, and mapping them to the specific entries in the resolution path.

It basically boils down to the question if we think it's beneficial to _allow_ adding targets dynamically at runtime...

Reply:
It just seems simpler in implementation and more flexible. I'm generally in favor of less opinionated solutions if they don't add complexity.
I could see a use case come up for extensions that want to put dependencies elsewhere, but they could also just write into var libs i suppose.
Maybe we can target it in a follow up PR once there are more requirements